enhancements/samples-generation.md

```diff
@@ -0,0 +1,613 @@
+
+## Motivation
+
+### Goals
+
+- Generate samples via Go to improve the maintainability, readability and reusability
+- Checking if samples are outdated via CI to reduce its risk of the sample projects gets outdated
+- Building samples on SDK to facilitate the development process to mock data and keep the samples maintained by using the same approach and design defined and used via the test
+- Re-using samples on tests to improve the maintainability and ensure that they are runnable.
+- Helpers and utilities available for tests and samples
+- Specific implementation following the same approach to define the sample and the tests context
+
+### Non-Goals
+
+- Link the samples in the docs or aligned then with the tutorials
+- Provide advanced helpers or Boilerplates to mock the data that could be used in the tests and/or in the samples generation
+- Push this implementation to upstream
+
+## Proposal
+
+### User Stories
+
+#### Generate samples via Go 
+
+- I am a maintainer/contributor, I want to be able to troubleshoot the code used to create the samples, so that I can easily fix the issues. 
+- I am a maintainer/contributor, I want to be develop the samples as I develop the tests, so that I do not need to be familiar with a different approach
+- I am a maintainer/contributor, I want to be easily create samples based on the tutorial steps, so that we can provide the expected result of the website tutorials
+
+#### Checking if samples are outdated via CI
+
+- I am a maintainer/contributor, I want to be able to check the samples update with the changes made, so that I do not need to spend effort on the samples after the release process
+- I am a maintainer/contributor, I want to able to check the samples updated in the diff of te github on the review process, so that I do not need to clone them locally
+
+#### Building samples on SDK
+
+- I am a maintainer/contributor, I want to be easily ensure that the changes made will not brake the steps required to build the projects, so that we can reduce the chance to introduce critical bugs. 
+- I am a maintainer/contributor, I want to check the result of the changes made in the scaffold by verifying the samples changes , so that I do not need to pull the pull request locally and run the steps to review the code. 
+- I am a maintainer/contributor, I want to have the samples update with the code changes, so that I do not update the samples project after the release
+- I am a maintainer/contributor, I want to have the samples update with the code changes, so that I do not update the samples project after the release
+- I am an operator developer, I want check the examples of projects and features provided, so that I can easily know how to build my operator projects
+- I am an operator developer, I want to check the final result of the tutorial steps, so that I can easily check if what I miss when the things do not go well as expected
```

Comment:
```suggestion
- I am an operator developer, I want to check the final result of the tutorial steps, so that I can easily compare if when the process does not go as expected
```